File: Java/chapter_15/app/src/main/java/com/glella/criminalintent/database/CrimeBaseHelper.java
package com.glella.criminalintent.database;

import android.content.Context;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;

import com.glella.criminalintent.database.CrimeDbSchema.Crimetable;

public class CrimeBaseHelper extends SQLiteOpenHelper {

    private static final int VERSION = 1;
    private static final String DATABASE_NAME = "crimeBase.db";

    public CrimeBaseHelper(Context context) {
        super(context, DATABASE_NAME, null, VERSION);
    }

    @Override
    public void onCreate(SQLiteDatabase db) {
        db.execSQL("create table " + Crimetable.NAME + "(" +
                " _id integer primary key autoincrement, " +
                Crimetable.Cols.UUID + ", " +
                Crimetable.Cols.TITLE + ", " +
                Crimetable.Cols.DATE + ", " +
                Crimetable.Cols.SOLVED + ", " +
                Crimetable.Cols.SUSPECT +
                ")"
        );
    }

    @Override
    public void onUpgrade(SQLiteDatabase sqLiteDatabase, int oldVersion, int newVersion) {
        //
    }
}

File: Kotlin/chapter_15_challenge/app/src/main/java/com/glella/criminalintent/database/CrimeBaseHelper.kt
package com.glella.criminalintent.database

import android.content.Context
import android.database.sqlite.SQLiteDatabase
import org.jetbrains.anko.db.ManagedSQLiteOpenHelper

class CrimeBaseHelper (context: Context) : ManagedSQLiteOpenHelper(context, CrimeDbSchema.Crimetable.DATABASE_NAME,null, CrimeDbSchema.Crimetable.VERSION ) {

    companion object {
        private var instance: CrimeBaseHelper? = null

        @Synchronized
        fun getInstance(ctx: Context): CrimeBaseHelper {
            if (instance == null) {
                instance = CrimeBaseHelper(ctx.applicationContext)
            }
            return instance!!
        }
    }


    override fun onCreate(db: SQLiteDatabase) {
        db.execSQL("create table " + CrimeDbSchema.Crimetable.NAME + "(" +
                " _id integer primary key autoincrement, " +
                CrimeDbSchema.Crimetable.Cols.UUID + ", " +
                CrimeDbSchema.Crimetable.Cols.TITLE + ", " +
                CrimeDbSchema.Crimetable.Cols.DATE + ", " +
                CrimeDbSchema.Crimetable.Cols.SOLVED + ", " +
                CrimeDbSchema.Crimetable.Cols.SUSPECT + ", " +
                CrimeDbSchema.Crimetable.Cols.SUSPECTID + ", " +
                CrimeDbSchema.Crimetable.Cols.SUSPECTPHONE +
                ")"
        )

    }

    override fun onUpgrade(db: SQLiteDatabase, oldVersion: Int, newVersion: Int) {
        // Here you can upgrade tables, as usual

    }

}


// Access property for Context
val Context.database: CrimeBaseHelper
    get() = CrimeBaseHelper.getInstance(applicationContext)


